### DISCUSSION TITLE
HP OfficeJet Pro 8710 not detected on NAPS2 7.1.0 for Linux

### DISCUSSION THREAD

**Discussion by montr0321** [cyanfish/naps2#201] [Support]
Hi!

NAPS2 7.1.0 for Linux doesn't detect my HP OfficeJet Pro 8710 printer/scanner (nothing in "select source" window). Running Debian Stable, installed from the .deb package (not flatpak).

Scanner is detected in other Linux frontends and by scanimage:

$ scanimage -L
> device `escl:http://localhost:60000' is a HP OfficeJet Pro 8710 [######] (USB) platen,adf scanner
> 
> device `airscan:e0:HP OfficeJet Pro 8710 [######] (USB)' is a eSCL HP OfficeJet Pro 8710 [######] (USB) ip=127.0.0.1
> 
> device `hpaio:/usb/HP_OfficeJet_Pro_8710?serial=[########]' is a Hewlett-Packard HP_OfficeJet_Pro_8710 all-in-one
> 
> device `hpaio:/net/hp_officejet_pro_8710?ip=127.0.0.1&queue=false' is a Hewlett-Packard hp_officejet_pro_8710 all-in-one

And same scanner is detected by NAPS2 for Windows when running Windows.

Any idea how to fix, or should I submit a bug report? Thank you!

**Comment by cyanfish** [comment]
Can you follow the instructions [here](https://www.naps2.com/doc/troubleshooting#error-log) to see if there are any error logs?

Also, to double check, are you waiting until the spinning progress indicator is gone? SANE drivers can sometimes take a while (10+ seconds) to load devices.

**Comment by montr0321** [comment]
Thank you very much for the reply! I was indeed not waiting long enough, took about 7 seconds for the entries (four of them for the same scanner, as in the scanimage -L result) to show up.

Which leads to 2 additional issues:

1) There was no spinning progress indicator during this time, so nothing to show that it was working. See screenshot: the gear wheel underneath the Cancel button was stationary, not spinning.

2) Of the four choices for this scanner, the bottom 2 (hpaio) don't work (SANE IO error), and the top one (escl) doesn't work either (always fails with "no pages are in the feeder," even if set to glass).

So the only choice that works is 2 (airscan). But it is very very slow. It hangs for about 25 seconds on "Scanning Page 1" every time I scan something from glass, at each new page (feeder works fine). I'm attaching a debug log from one such scan (glass, that took an extra 25 seconds), but it doesn't seem to show the delay. There is no error log, just the debug log.

Thank you very much if you have any ideas that might help!

[debuglog.txt](https://github.com/cyanfish/naps2/files/12797474/debuglog.txt)
![Screenshot-Naps2](https://github.com/cyanfish/naps2/assets/146587386/1a58672a-bb3a-481c-acef-bef05e07a79f)

**Comment by montr0321** [comment]
Adding: this problem (25-second wait before each scan) doesn't happen using scanimage on the command line, or the other Linux frontend I tried.

**Reply by cyanfish** [reply]
Do the other options work with scanimage? If so can you provide a debuglog for them?

**Reply by montr0321** [reply]
No, the other options fail on scanimage with the same error messages as on NAPS (empty feeder for escl, SANE IO error for hpaio).

So the airscan one is the only one that works on any program, but it's the one that has the 25-second delay for each page on NAPS.
